File: mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
//Get positive number from 1 to 8
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

//Build left pyramid
    /*for (int i = 0; i < height; i++)
    {
      for (int j = -1; j < i; j++)
        {
            printf("#");
        }
        printf("\n");
    }*/

//Build right pyramid with dots
    /*for (int line = 0; line < height; line++)
    {
        for (int dots = 0; dots < height - line - 1; dots++)
        {
            printf(" ");
        }
        for (int hash = 0; hash <= line; hash++)
        {
            printf("#");
        }
        printf("\n");
    }*/

//Build difficult Mario pyramid
    for (int line = 0; line < height; line++)
    {
        for (int dots = 0; dots < height - line - 1; dots++)
        {
            printf(" ");
        }
        for (int hash = 0; hash <= line; hash++)
        {
            printf("#");
        }
        printf("  ");
        for (int hash2 = 0; hash2 <= line; hash2++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: sentimental-mario-more/mario.py
from cs50 import get_int


def main():
    height = get_height()
    for line in range(height):
        for dots in range(height - line - 1):
            print(" ", end="")  # Print ' ' without new line in the end
        for hash in range(line + 1):
            print("#", end="")  # Print '#' without new line in the end
        print("  ", end="")
        for hash2 in range(line + 1):
            print("#", end="")  # Print '#' without new line in the end
        print()  # Print new line


# Get a positive integer between 1 and 8 from user in a loop
def get_height():
    while True:
        n = get_int("Enter a positive integer between 1 and 8 inclusive: ")
        if 1 <= n <= 8:
            return n


if __name__ == "__main__":
    main()
